chat-android/src/main/java/com/ably/chat/Typing.kt

```diff
@@ -163,82 +187,132 @@ internal class DefaultTyping(
     override suspend fun get(): Set<String> {
         logger.trace("DefaultTyping.get()")
         room.ensureAttached(logger) // CHA-T2d, CHA-T2c, CHA-T2g
-        return channelWrapper.presence.getCoroutine().map { it.clientId }.toSet()
+        return currentlyTypingMembers
     }
 
     override suspend fun start() {
         logger.trace("DefaultTyping.start()")
-
-        typingScope.launch {
-            // If the user is already typing, reset the timer
-            if (typingJob != null) {
-                logger.debug("DefaultTyping.start(); already typing, resetting timer")
-                typingJob?.cancel()
-                startTypingTimer()
-            } else {
-                startTypingTimer()
-                room.ensureAttached(logger) // CHA-T4a1, CHA-T4a3, CHA-T4a4
-                channelWrapper.presence.enterClientCoroutine(room.clientId)
+        typingScope.async {
```

Comment:
`launch` fits better here

Reply:
Yeah, actually I have seen cases where `join` doesn't throw exceptions for inside code. 
Will double check and get back 👍

Reply:
For reference, I have created a PR -> https://github.com/ably/ably-chat-kotlin/pull/120/files. 
You can clone at the branch and re-run the tests.

Reply:
Our requirement is for `start`/ `keyStroke` and `stop` method to throw exceptions if anything goes wrong.
With `launch` and `join` no exceptions are captured : (